src/all/comicskingdom/src/eu/kanade/tachiyomi/extension/all/comicskingdom/ComicsKingdom.kt

```diff
@@ -0,0 +1,285 @@
+package eu.kanade.tachiyomi.extension.all.comicskingdom
+
+import android.app.Application
+import android.content.SharedPreferences
+import android.util.Log
+import androidx.preference.PreferenceScreen
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+import uy.kohesive.injekt.injectLazy
+import java.text.SimpleDateFormat
+import java.util.Locale
+import kotlin.math.ceil
+import kotlin.math.roundToInt
+
+class ComicsKingdom(override val lang: String) : ConfigurableSource, ParsedHttpSource() {
+
+    private val json: Json by injectLazy()
+    private val langText = if (lang == "en") "comics" else "spanish"
+    private val dateFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss", Locale.getDefault())
+    private val compactChapterCountRegex = Regex("\"totalItems\":(\\d+)")
+    private val urlBase = "comicskingdom.com"
+
+    override val name = "Comics Kingdom"
+
+    override val baseUrl = "https://$urlBase"
+
+    override val supportsLatest = true
+
+    override fun popularMangaSelector() = ".comic-result-container .comic-card"
+    override fun latestUpdatesSelector() = popularMangaSelector()
+    override fun searchMangaSelector() =
+        "h2:contains(Comic Titles) ~ div .posts-container .comic-card"
+
+    override fun popularMangaNextPageSelector() = ".ck-pagination__item--next"
+    override fun latestUpdatesNextPageSelector() = popularMangaNextPageSelector()
+    override fun searchMangaNextPageSelector() =
+        "h2:contains(Comic Titles) ~ div .ck-pagination__item--next"
+
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/$langText?page=$page", headers)
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        return GET("$baseUrl/$langText?page=$page&sortby=newest", headers)
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request =
+        GET(
+            "$baseUrl/search".toHttpUrl().newBuilder().apply {
+                addQueryParameter("key", query)
+                // addQueryParameter("page", page.toString()) for singular strips
```

Comment:
you can just specify difference in query parameter in comment, also is it not being used?